File: scratch/merge_css.py
import os

TAILADMIN_CSS_PATH = r"d:\SeCard\free-nextjs-admin-dashboard-main\free-nextjs-admin-dashboard-main\src\app\globals.css"
ORIGINAL_CSS_PATH = r"d:\SeCard\src\app\globals.css"

with open(TAILADMIN_CSS_PATH, "r", encoding="utf-8") as f:
    tailadmin_css = f.read()

with open(ORIGINAL_CSS_PATH, "r", encoding="utf-8") as f:
    original_css = f.read()

# If the custom overrides block already exists in original_css, let's extract just that part
custom_separator = "/* ================= CUSTOM ECARD OVERRIDES ================= */"
if custom_separator in original_css:
    custom_part = original_css.split(custom_separator)[1].strip()
else:
    custom_part = original_css

# The original font imports:
font_imports = "@import url('https://fonts.googleapis.com/css2?family=Be+Vietnam+Pro:wght@300;400;500;600;700;800;900&family=Inter:wght@300;400;500;600;700;800;900&display=swap');\n\n"

combined_css = font_imports + tailadmin_css + "\n\n/* ================= CUSTOM ECARD OVERRIDES ================= */\n\n" + custom_part

with open(ORIGINAL_CSS_PATH, "w", encoding="utf-8") as f:
    f.write(combined_css)

print("CSS Merged Successfully!")

File: scratch/merge_css.js
const fs = require('fs');
const path = require('path');

const tailadminCssPath = path.resolve('free-nextjs-admin-dashboard-main/free-nextjs-admin-dashboard-main/src/app/globals.css');
const originalCssPath = path.resolve('src/app/globals.css');

if (!fs.existsSync(tailadminCssPath)) {
    console.error('TailAdmin CSS file not found at:', tailadminCssPath);
    process.exit(1);
}

if (!fs.existsSync(originalCssPath)) {
    console.error('Original CSS file not found at:', originalCssPath);
    process.exit(1);
}

const tailadminCss = fs.readFileSync(tailadminCssPath, 'utf8');
const originalCss = fs.readFileSync(originalCssPath, 'utf8');

const customSeparator = "/* ================= CUSTOM ECARD OVERRIDES ================= */";
let customPart = originalCss;
if (originalCss.includes(customSeparator)) {
    customPart = originalCss.split(customSeparator)[1].trim();
}

const fontImports = "@import url('https://fonts.googleapis.com/css2?family=Be+Vietnam+Pro:wght@300;400;500;600;700;800;900&family=Inter:wght@300;400;500;600;700;800;900&display=swap');\n\n";

const combinedCss = fontImports + tailadminCss + "\n\n" + customSeparator + "\n\n" + customPart;

fs.writeFileSync(originalCssPath, combinedCss, 'utf8');
console.log("CSS Merged Successfully!");
